examples/cpp/comm/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ add_subdirectory (event)
 add_subdirectory (generic)
 add_subdirectory (layer)
 add_subdirectory (logging)
+add_subdirectory (multi-trans-units)
```

Comment:
We shouldn't have an explicit multiple translation unit example like this. What I wrote was just to reproduce the reported problem. We should just take one of these existing examples and split it across translation units. E.g., implement a service separate from `main()`.